app/reflexes/products_reflex.rb

```diff
@@ -38,8 +38,8 @@ def bulk_update
     @products = product_set.collection # use instance variable mainly for testing
 
     if product_set.save
-      # flash[:success] = with_locale { I18n.t('.success') }
-      # morph_admin_flashes  # ERROR: selector morph type has already been set
+      flash[:success] = I18n.t('admin.products_v3.bulk_update.success')
+      broadcast_admin_flashes
```

Comment:
StimulusReflex can only have one `morph` , which is analogous to Rails' `render`. Because the main action (render product list) is morphed in a separate container, we need to send two chunks of HTML rather than the one full-page document. 
It's kinda ugly though and I'd prefer if the two chunks could be sent in the one response.

🤔 Which begs the question, why not just use a standard rails controller action and render the full page?.. I'll probably try that in the next PR..

Reply:
That makes sense, in a way. But it also seems like an unnecessary limitation of the framework.

So I guess that we have two situations in which flash messages appear:

1. You trigger an action and the flash message is the only feedback you get. A morph is perfect then.
2. The page changes and you see a flash message. Some would argue that you don't need a flash message when you see the result on the page anyway. But they are especially useful for errors, in which case we may not want to re-render the page?

Anyway, given that our current UX design asks for page changes and flashes at the same time, we need to do an HTML replace. And it's not the end of the world. Why not.

The only other idea I have is to add a flash message to the HTML of the main part that is morphed, like you have error messages under input fields. So instead of a global `#flashes` container, you can have a `.flash` element anywhere on the page. And we could set up CSS in a way that the messages appear and disappear automatically. We could even use the new modal element for that, I think. The only downside of this approach would be that we need to *merge* HTML pieces and that multiple flash elements would probably overlap and can't be shown next to each other because they are not siblings. So maybe that's an error-prone approach. Ignore for now.

> Which begs the question, why not just use a standard rails controller action and render the full page?

If you pretty much render the whole page anyway, then yes, you can just do it in the controller. But if you just want to re-render the table without the search input fields then we should only render that.

Reply:
Some great points Maikel! Thanks for thinking outside the box. I don't think we need to support multiple messages, in fact I think that would be bad. Our "flash" model should probably enforce that.

So I actually think your suggestion is a good idea. I gave it a try and it works, see new commits. I'm not 100%  happy with it, because it means flash functionality is no longer provided by application helpers, instead you need to know to include the partial. Maybe we need a helper `morph_with_flashes` that appends flashes for you. What do you think of that?

Reply:
Sounds good. I do remember though that I reviewed a PR once that was enabling multiple flash messages. That may have been replaced later by a summary again. I can't remember that part. :wink: 

Maybe something like bulk operation failing on some items and showing a flash for each?

Reply:
Ha, maybe. But I don't think it would be good if there were multiple messages. What if you updated 50 products at once?

The flash object only holds one "success" at a time (for example). I can see that Matt did  update it to include an `.each`, but this just means we can show different types together (eg success and notice). But I can't imagine we would want to do that.